IO-phone/src/app/poll/poll.component.ts

```diff
@@ -1,20 +1,110 @@
 import { Component, OnInit, inject } from '@angular/core';
 import { PollService } from './poll.service';
+import { Answer, Poll, Question, Student } from './model';
+import { CommonModule } from '@angular/common';
+import { MatButtonModule } from "@angular/material/button";
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatInputModule } from '@angular/material/input';
+import { FormGroup, FormBuilder, Validators, ReactiveFormsModule } from '@angular/forms';
 
 @Component({
   selector: 'app-poll',
   standalone: true,
-  imports: [],
+  imports: [
+    CommonModule, ReactiveFormsModule,
+    MatButtonModule, MatFormFieldModule, MatInputModule, MatInputModule
+  ],
   templateUrl: './poll.component.html',
   styleUrl: './poll.component.css'
 })
-export class PollComponent implements OnInit {
+export class PollComponent {
   private pollService = inject(PollService);
+  private formBuilder = inject(FormBuilder);
+
+  pollTemplate: Poll | null = null;
+  questions: Question[] = [];
+
+  students: Student[] = [];
+  currentStudentIndex = -1;
+
+  pollForm: FormGroup = this.formBuilder.group({});
+
+  constructor() {
+    this.pollService.students$.subscribe(students => {
+      this.students = students;
+      this.currentStudentIndex = 0;
+    })
+    
+    this.pollService.pollTemplate$.subscribe(template => {
+      let questions = [];
+
+      // questions must be mapped to array, or the their original order won't be preserved because of the keyvalue pipe 
+      for (let [_key, value] of Object.entries(template.questions)) {
```

Comment:
Tutaj jest taka śmieszna sprawa, że o ile 99% przeglądarek implementuje obiekty w JS tak, że kolejność jest zachowywana, o tyle nie jest to zapewniane w żaden sposób przez standard.

Jeżeli chcielibyśmy to zrobić zgodnie ze sztuką, to trzebaby otypować `QuestionMap` zamiast:
```
export interface QuestionMap {
    readonly [key: string]: Question;
}
```
to jakoś tak:
```
export type QuestionMap = {
    readonly key: string;
    readonly value: Question;
}[]
```
Natomiast tego już nie zmieniaj, bo to i tak będzie w praktyce zawsze działać XD